pkg/loki/modules.go

```diff
@@ -1753,7 +1753,7 @@ func (t *Loki) initAnalytics() (services.Service, error) {
 
 // The Ingest Partition Ring is responsible for watching the available ingesters and assigning partitions to incoming requests.
 func (t *Loki) initPartitionRing() (services.Service, error) {
-	if !t.Cfg.Ingester.KafkaIngestion.Enabled {
+	if !t.Cfg.Ingester.KafkaIngestion.Enabled && !t.Cfg.Distributor.KafkaEnabled {
```

Comment:
This makes sense but the distributor won't be able to send any data to Kafka unless the ingester is also enabled due to needing some active partitions. I wonder if we should just use a single flag to enable the kafka write path?

Reply:
I understand although it feels weird that having `t.Cfg.Distributor.KafkaEnabled` without `t.Cfg.Ingester.KafkaIngestion.Enabled` panics ? Should we validate instead ?

Reply:
Yep, I think we need to validate that ingestion is enabled if distributor is enabled, since they're coupled in the architecture.

Reply:
To clarify, I think we want to keep both flags because we want the option to stop writing to Kafka without changing the ingesters.
But we also want a validation for `All` (and maybe `Write`) mode to enforce that the ingester is enabled if the distributor is enabled. We don't want this check for micro-services mode though, since those flags can be set explicitly on those components.